server/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -361,6 +360,19 @@ public String details() {
         return message + (failure == null ? "" : ", failure " + ExceptionsHelper.stackTrace(failure));
     }
 
+    /**
+     * Returns the number of shards that are unassigned and currently being delayed.
+     */
+    public static int numberOfDelayedUnassigned(ClusterState state) {
+        int count = 0;
+        for (ShardRouting shard : state.getRoutingNodes().unassigned()) {
+            if (shard.unassignedInfo().delayed()) {
+                count++;
+            }
+        }
+        return count;
+    }
```

Comment:
Lets move this into RoutingNodes. 
And `findNextDelayedAllocation` to DelayedAllocationService (probably in the separate pr). 
There is no need for UnassignedInfo to depend on the ClusterState.